### PR TITLE
Merged tests for Binary Search Recursive and Non-Recursive

### PR DESCRIPTION
**Pull request by soummyaah**
Refs #44 
@aktech Please check

- Team l1ghtsab3r

### CODE REVIEW

**Comment by aktech** [comment]
@soummyaah You need to fix the conflicts.

**Comment by soummyaah** [comment]
@aktech Please review.

File: pydsa/tests/test_binary_search.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
 from pydsa.binary_search import binary_search
 from random import randint
 
+
```
Comment (CuriousLearner) [inline pydsa/tests/test_binary_search.py]: Why this blank line?

Comment (soummyaah) [inline pydsa/tests/test_binary_search.py]: According to pep8 conventions there should be two blank lines between import statements and code.

**Comment by CuriousLearner** [comment]
LGTM! @aktech Would have a look!

**Comment by CuriousLearner** [comment]
@soummyaah Please create a separate function for binary_search_recursive

**Comment by soummyaah** [comment]
So, just to confirm. Same file but different functions?

**Comment by CuriousLearner** [comment]
@soummyaah Yes, there would be two functions in same file, one for iterative and another for recursive.

**Comment by soummyaah** [comment]
@aktech @CuriousLearner Please review.

**Comment by aktech** [comment]
Looks good to me, Thanks for working on this.

**Pull request merged**